cmd/orgIdColumnPopulator/main.go

```diff
@@ -0,0 +1,197 @@
+
+			err = orgIdColumnPopulator.MapAccountToOrgId(context.TODO(), db, dbTable, dbAccountColumn, dbOrgIdColumn, dbOperationTimeout, batchSize, accountNumberTranslator, metricsRecorder)
+			if err != nil {
+				log.Fatal(err)
+			}
+		},
+	}
+
+	rootCmd.Flags().StringVarP(&dbConnMetadata.host, "db-host", "l", "localhost", "Database hostname")
+	rootCmd.Flags().IntVarP(&dbConnMetadata.port, "db-port", "p", 5432, "Database port")
+	rootCmd.Flags().StringVarP(&dbConnMetadata.user, "db-user", "u", "insights", "Database user")
+	rootCmd.Flags().StringVarP(&dbConnMetadata.password, "db-password", "w", "1234", "Database password")
+	rootCmd.Flags().StringVarP(&dbConnMetadata.name, "db-name", "n", "cloud-connector", "Database name")
+	rootCmd.Flags().StringVarP(&dbConnMetadata.sslMode, "db-ssl-mode", "s", "disable", "Database ssl mode")
+
+	rootCmd.Flags().StringVarP(&dbTable, "db-table-name", "t", "connections", "Database name")
+	rootCmd.Flags().StringVarP(&dbAccountColumn, "db-account-column-name", "a", "account", "Account column within db table")
+	rootCmd.Flags().StringVarP(&dbOrgIdColumn, "db-org-id-column-name", "o", "org-id", "OrgID column within db table")
+	rootCmd.Flags().IntVar(&dbOperationTimeout, "db-operation-timeout", 5, "Timeout for each db operation in number of seconds")
+
+	rootCmd.Flags().BoolVarP(&clowderConfig, "read-clowder-config", "C", false, "Read db config from clowder config file")
+
+	rootCmd.Flags().IntVarP(&batchSize, "batch-size", "b", 500, "Number of accounts to retrieve from the database at once ")
+
+	rootCmd.Flags().StringVarP(&prometheusPushGatewayAddr, "prometheus-push-addr", "g", "", "Address of prometheus push gateway")
+
+	rootCmd.Flags().StringVarP(&translatorServiceAddr, "ean-translator-addr", "T", "", "Address of EAN translator service")
+	rootCmd.Flags().IntVarP(&translatorServiceTimeout, "ean-translator-timeout", "O", 10, "Timeout for calling the EAN translator service")
+
+	return rootCmd
+}
+
+type dbConnectionMetadata struct {
+	host     string
+	port     int
+	user     string
+	password string
+	name     string
+	sslMode  string
+}
+
+func buildDBConnectionString(metadata dbConnectionMetadata) string {
+	connStr := fmt.Sprintf("host=%s port=%d user=%s password=%s dbname=%s sslmode=%s",
+		metadata.host,
+		metadata.port,
+		metadata.user,
+		metadata.password,
+		metadata.name,
+		metadata.sslMode)
+	return connStr
+}
+
+func initializeDBConnection(connStr string) (*sql.DB, error) {
+	// FIXME: Remove me
+	log.Println("connStr: ", connStr)
+
+	db, err := sql.Open("postgres", connStr)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return db, nil
+}
+
+func buildAccountNumberTranslator(translatorServiceAddr string, timeout int) tenantid.BatchTranslator {
+
+	if translatorServiceAddr != "" {
+		log.Println("Using real EAN translator impl")
+		option := tenantid.WithTimeout(time.Duration(timeout) * time.Second)
+		return tenantid.NewTranslator(translatorServiceAddr, option)
```

Comment:
idea: if push gateway address is defined then also enable translator client metrics (`WithMetrics(registry)`) and report them to push gateway

Reply:
Great idea.  I have never looked into the prometheus registry code.  I think I have this working but I don't have a push gateway to actually test this against.  Maybe I could stand one up locally using docker/podman?  If I print out the registry object just before the Push() call, then it looks like the metrics from the translator code are in a map within the registry object....so I think its working as expected.